src/main/java/core/basesyntax/service/strategy/BalanceOperation.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+    private FruitStorageDao storageDao;
+
+    public BalanceOperation(FruitStorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        storageDao.getAllFruits().put(transaction.getFruit(), transaction.getQuantity());
+    }
+}
```

Comment:
The `BalanceOperation` class directly modifies the storage by calling `storageDao.getAllFruits().put(...)`. This can lead to a violation of the encapsulation principle. It's better to have a method in the DAO that abstracts the details of how the balance is updated, ensuring that the storage can only be modified in controlled ways.